### ISSUE TITLE
new RPC - ListTrades

### ISSUE THREAD

**Issue by ImmanuelSegol** [ExchangeUnion/xud#677]
We should have an RPC method to return all trades from the database.

Should be implemented after #608 is merged.

**Comment by sangaman** [comment]
I'm not so sure about *all* trades - that would quickly become a ton of data and I'm not sure all of it is needed. Listing recent trades makes sense, but I think for crunching long term data we should look at some reporting that works outside of `xud` itself (which shouldn't be a priority for now).

**Comment by kilrau** [comment]
> Once this is in place, I'll create a separate issue to implement a `ListTrades` RPC method.
> 
> I also think we might want a `LookupOrder` method which checks if an order is still live or, if not, tries to find record of it in `trades` or `swapdeals`. Also something for a separate issue and PR.

Definitely not *all* trades. Also I checked several accounts I have with exchanges: it's usually 'order book' and 'orders' (which is **my** closed and open orders). But I don't think any of us is up for another renaming.

Suggestion: `listTrades` comes in two sections: `Open` and `Closed`. Default should limit to the last e.g. 15, `all` flag gets all and like Daniel said  `reporting` is not our focus rn

**Comment by kilrau** [comment]
Related: https://github.com/ExchangeUnion/xud/issues/681

**Comment by kilrau** [comment]
Related: ListOrderHistory (https://github.com/ExchangeUnion/xud/issues/689)
